flink-streaming-java/src/main/java/org/apache/flink/streaming/api/lineage/LineageUtils.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.lineage;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.connector.source.Boundedness;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Util class for creating default LineageDataset and LineageVertex. */
+@Internal
+public class LineageUtils {
+
+    public static LineageDataset datasetOf(
+            String name, String namespace, TypeDatasetFacet typeDatasetFacet) {
+        return datasetOf(name, namespace, Collections.singletonList(typeDatasetFacet));
+    }
+
+    public static LineageDataset datasetOf(
```

Comment:
I wonder if these datasetOf constructor methods can be put in DefaultLineageDataset or a class that creates DefaultLineageDataset , maybe a DefaultLineageDatasetProvider

Reply:
It is mainly to make the LineageDataset creation easier and simplify the code in each of connectors.